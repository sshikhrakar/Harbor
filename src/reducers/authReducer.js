import {
  LOGIN_VIA_EMAIL_FULFILLED,
  SIGNUP_VIA_EMAIL_FULFILLED,
  REGISTER_FCM_TOKEN,
} from '../actions/actionTypes';

export const initialState = {
  isLoggedIn: false,
  fcmTokenRegistered: false,
};

/**
 * Reducer for authenticaion state.
 *
 * @param {Object} state=initialState
 * @param {Object} action
 * @returns {Object}
 */
function auth(state = initialState, action) {
  switch(action.type) {

    case LOGIN_VIA_EMAIL_FULFILLED:
      return {
        ...state,
        isLoggedIn: true,
      };

    case SIGNUP_VIA_EMAIL_FULFILLED:
      return {
        ...state,
        isLoggedIn: true,
      };

    case REGISTER_FCM_TOKEN:
      return {
        ...state,
        fcmTokenRegistered: true,
      };

    default:
      return state;
  }
}

export default auth;
