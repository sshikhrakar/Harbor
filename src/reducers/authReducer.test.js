import authReducer, { initialState } from './authReducer';

describe('REDUCER: AUTH', () => {

  it('should return initialState when state is undefined and action is empty', () => {
    expect(authReducer(undefined, {})).toEqual(initialState);
  });

});
