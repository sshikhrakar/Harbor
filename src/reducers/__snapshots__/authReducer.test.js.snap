// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REDUCER: AUTH should set logged in to true when signup is successful 1`] = `
Object {
  "fcmTokenRegistered": false,
  "isLoggedIn": true,
}
`;
